pycsw/core/util.py

```diff
@@ -284,7 +288,7 @@ def transform_mappings(queryables, typename, reverse=False):
     if reverse:  # from csw:Record
         for qbl in queryables.keys():
             if qbl in typename.values():
-                tmp = [k for k, v in typename.iteritems() if v == qbl][0]
+                tmp = [k for k, v in typename.items() if v == qbl][0]
```

Comment:
The trouble with non-guaranteed order is sometimes it will work and sometimes it won't.